src/error/metadata/mod.rs

```diff
@@ -145,33 +145,12 @@ impl From<&mut anyhow::Error> for RoverErrorMetadata {
                     Some(RoverErrorSuggestion::FixContractPublishErrors),
                     Some(RoverErrorCode::E040),
                 ),
-                RoverClientError::OperationCheckFailure {
-                    graph_ref,
+                RoverClientError::CheckWorkflowFailure {
+                    graph_ref: _,
                     check_response: _,
                 } => (
-                    Some(RoverErrorSuggestion::FixOperationsInSchema {
-                        graph_ref: graph_ref.clone(),
-                    }),
-                    Some(RoverErrorCode::E030),
-                ),
-                RoverClientError::DownstreamCheckFailure {
-                    blocking_downstream_variants: _,
-                    target_url,
-                } => (
-                    Some(RoverErrorSuggestion::FixDownstreamCheckFailure {
-                        target_url: target_url.clone(),
-                    }),
-                    Some(RoverErrorCode::E037),
-                ),
-                RoverClientError::OtherCheckTaskFailure {
-                    has_build_task: _,
-                    has_downstream_task: _,
-                    target_url,
-                } => (
-                    Some(RoverErrorSuggestion::FixOtherCheckTaskFailure {
-                        target_url: target_url.clone(),
-                    }),
-                    Some(RoverErrorCode::E036),
```

Comment:
I think we still want to be able to return these error types from checks, what do you think?

Reply:
The refactor made in this PR removes any of these from returning, instead a new code (042) to encapsulate the general case of a failing check was added. I can keep these but nothing will use or return them at least based on the current changes in this PR.

Reply:
Right, understood that you've encapsulated them all into a single error. It's probably good enough to do that and just not emit these errors anymore, folks just won't have the additional info of _what_ exactly failed without looking elsewhere in the error. This is probably fine.